src/wallet/db.cpp

```diff
@@ -71,6 +70,24 @@ BerkeleyEnvironment* GetWalletEnv(const fs::path& wallet_path, std::string& data
         env_directory = wallet_path;
         database_filename = "wallet.dat";
     }
+}
+
+bool IsWalletLoaded(const fs::path& wallet_path)
+{
+    fs::path env_directory;
+    std::string database_filename;
+    SplitWalletPath(wallet_path, env_directory, database_filename);
+    LOCK(cs_db);
+    auto env = g_dbenvs.find(env_directory.string());
+    if (env == g_dbenvs.end()) return false;
+    auto db = env->second.m_databases.find(database_filename);
```

Comment:
I think this is leaking the implementation detail of `m_databases`, wouldn't checking whether a database exists ideally be a responsibility of the database environment. e.g. add a `IsDatabaseLoaded` on `DBenv`?

Reply:
Note that `BerkeleyEnvironment::mapDb` is currently used outside. I consider this a bugfix so any improvement/refactor could come next.

Reply:
Resolved by adding `IsDatabaseLoaded`